### PR TITLE
Docs: IC Identity for devs and other updates

### PR DESCRIPTION
**Pull request by Eve832**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by Dfinity-Bjoern — CHANGES_REQUESTED** [review]
I don't think it is using the correct terminology w.r.t. identity, pseudonym etc.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 1]
```diff
@@ -6,21 +6,23 @@
...
+Unlike most authentication methods, your Internet Identity does not require you to set and manage passwords or provide any personal identifying information to applications or to Internet Identity. Instead, you register a device by using any of the device's own authentication methods. If the device supports WebAuthn, and accepts your authentication method, a public key is created and is represented by an ID number (you User Number). At the same time, the security chip in your device generates a  private key, which creates a signature associated with your ID number. The private key is secret and never leaves your device. No one, not even you, can see the private key. The unique signature from your private key is what Internet Identity uses to prove that you are in possession of the correct private key for your ID number.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:11]: Dom explicitly wanted to drop the term "user number".

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 2]
```diff
@@ -6,21 +6,23 @@
...
 
-NOTE: On macOS/iOS/ipadOS (including iPhones), authentication using Safari is coupled to your browser profile. If you want to log in to an application in a different browser, or if you use multiple Safari browser profiles, you have to add the combination of your authentication method and the new browser as a new device. See: <<Add a device,`+Add a device+`>>.
+When you access an application that uses Internet Identity, you will be re-directed to the Internet Identity app. Once you authenticate into the app, Internet Identity creates a unique identity that is saved in the application's persistance layer (or web server), but does not contain any personal information, because at no point have you offered any personal information. However, this unique and anonymous identity is associated with your public ID number, so if you lose the number, you will also lose the account identity associated with it.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:13]: I don't understand what you mean by "that is saved in the application's persistence layer" but we will definitely want to drop the term "web server" there.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 3]
```diff
@@ -6,21 +6,23 @@
...
 
-Windows users can only use a security key as an authorized device.
+NOTE: On OSX (including iPhones), authentication using Safari is coupled to your browser profile. If you want to log in to an application in a different browser, or if you use multiple Safari browser profiles, you have to add the combination of your authentication method and the new browser as a new device. See: <<Add a device,`+Add a device+`>>.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:15]: "OS X" was the correct name until last year.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 4]
```diff
@@ -6,21 +6,23 @@
...
+
+Windows Hello users can only use a security key as an authorized device.
 
 Firefox does not currently accept OSX with any device authentication method other than a security key. 
 
 == Create an Internet Identity
 
-You can securely access applications running on the {IC} that use Internet Identity for authentication, by selecting an authentication method and having an Internet Identity ID number generated for you. You can create as many identities as you want by registering new ID numbers. 
+You can securely access applications running on the {IC} that use the Internet Identity for authentication, by selecting an authentication method and having an Internet Identity ID number (user number) generated for you. You can create as many identities as you want by registering new ID numbers.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:23]: Drop user number.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 5]
```diff
@@ -6,21 +6,23 @@
...
 
 Firefox does not currently accept OSX with any device authentication method other than a security key. 
 
 == Create an Internet Identity
 
-You can securely access applications running on the {IC} that use Internet Identity for authentication, by selecting an authentication method and having an Internet Identity ID number generated for you. You can create as many identities as you want by registering new ID numbers. 
+You can securely access applications running on the {IC} that use the Internet Identity for authentication, by selecting an authentication method and having an Internet Identity ID number (user number) generated for you. You can create as many identities as you want by registering new ID numbers. 
 
-When you first access an application running on the {IC}, you are directed to Internet Identity and asked to enter a ID number to log in. If you are a new user, you need to register to obtain an ID number.
+When you first access an application running on the {IC}, you are directed to Internet Identity and asked to enter a User Number to log in. If you are a new user, you need to register to obtain a user number.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:25]: User number is deprecated and is now named "ID number".

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 6]
```diff
@@ -48,14 +50,14 @@ Authenticate using the method you selected when prompted.
 Your identity is not created until you perform this step. 
 +
 At this point, depending on the device you are using, you might be asked to either use your device authenitication method, or to use an identity saved to your security key. If you are registering for the first time, choose to use the device authentication. 
-. Save your *ID number*.
+. Save your *User Number*.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:53]: See above.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 7]
```diff
@@ -48,14 +50,14 @@ Authenticate using the method you selected when prompted.
...
 +
-When you have registered your device, you’ll receive an *ID number*.
+When you have registered your device, you’ll receive a *User Number*.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:55]: See above.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 8]
```diff
@@ -48,14 +50,14 @@ Authenticate using the method you selected when prompted.
...
 At this point, depending on the device you are using, you might be asked to either use your device authenitication method, or to use an identity saved to your security key. If you are registering for the first time, choose to use the device authentication. 
-. Save your *ID number*.
+. Save your *User Number*.
 +
-When you have registered your device, you’ll receive an *ID number*.
+When you have registered your device, you’ll receive a *User Number*. 
 +
 This number is unique, but it is not a secret, so you should save the number in multiple places. 
-Your browser will remember your ID number, but you will need it if you log in on a different computer, change your browser profile, or if you clear your browser state. 
+Your browser will remember your user number, but you will need it if you log in on a different computer, change your browser profile, or if you clear your browser state.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:58]: See above.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 9]
```diff
@@ -48,14 +50,14 @@ Authenticate using the method you selected when prompted.
...
 +
-When you have registered your device, you’ll receive an *ID number*.
+When you have registered your device, you’ll receive a *User Number*. 
 +
 This number is unique, but it is not a secret, so you should save the number in multiple places. 
-Your browser will remember your ID number, but you will need it if you log in on a different computer, change your browser profile, or if you clear your browser state. 
+Your browser will remember your user number, but you will need it if you log in on a different computer, change your browser profile, or if you clear your browser state. 
 +
-WARNING: If you lose your ID number and are logged out on all devices, you cannot log in to Internet Identity to manage your devices, or access any of your applications, unless you have set up account recovery in the next step.
+WARNING: If you lose your user number and are logged out on all devices, you cannot log in to Internet Identity to manage your devices, or access any of your applications, unless you have set up account recovery in the next step.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:60]: See above.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 10]
```diff
@@ -67,11 +69,13 @@ On the next screen, you can select one of the following options:
 
 * *Seed Phrase*
 + 
-Select this option to generate a cryptographically-secure seed-phrase that you can use to recover your account. Make sure you save this phrase somewhere safe and known only to you. *Note that the first string in your seed phrase is your ID number*. You will need this number to begin the the recovery process.
+Select this option to generate a cryptographically-secure seed-phrase that you can use to recover your account. Make sure you save this phrase somewhere safe and known only to you. *Note that the first string in your seed phrase is your user number*. You will need this number to begin the the recovery process.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:72]: See above.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 11]
```diff
@@ -96,7 +100,7 @@ To register your phone as an additional authentication method using the phone's
 
 . Click *Already registered but using a new device?*
 
-. Enter your ID number and click *Continue*.
+. Enter your user ID number and click *Continue*.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:103]: See above. No "user" here.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 12]
```diff
@@ -110,20 +114,20 @@ NOTE: To use the screen lock option, you have to have screen lock activated on y
 . Authorize your phone.
 +
 After you’ve unlocked your phone, you will be provided with a URL and a QR code. You must use the URL or QR code in a browser in the computer that has already been authorized. For example, you can copy the URL and email it to yourself, then paste it into a browser on the computer. 
-. Enter your ID Number and click *Login*.
+. Enter your user ID number and click *Login*.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:117]: Drop "user".

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 13]
```diff
@@ -110,20 +114,20 @@ NOTE: To use the screen lock option, you have to have screen lock activated on y
...
 . Link your phone to your identity. 
 +
 If you’re sure that the link you pasted in the browser came from you, click *Yes, link new device*.
 . Give the device profile a name and click *Link Device*.
 +
-Your phone will be redirected to the login page, and you can now use it with your ID number to log in.
+Your phone will be redirected to the login page, and you can now use it with your user number to log in.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:123]: User number -> ID number.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc [hunk 14]
```diff
@@ -110,20 +114,20 @@ NOTE: To use the screen lock option, you have to have screen lock activated on y
...
 
 NOTE: You should register as many devices as possible to prevent you from losing access to your applications should you lose a device. Again, the best way to prevent accidental loss is to set up a recovery method. 
 
 
 == Recover a lost identity
 When you register an identity, you will be prompted to copy a cryptographically-secure seed phrase or to add a dedicated security key for recovery purposes. 
 
-You can choose to do this at any time, but note that if you lose your ID number or if you no longer have access to authorized devices, you will need the seed-phrase or the recovery security key to recover your identity. Without one of these, you will be locked out of any applications that require the associated identity. 
+You can choose to do this at any time, but note that if you lose your user number or if you no longer have access to authorized devices, you will need the seed-phrase or the recovery security key to recover your identity. Without one of these, you will be locked out of any applications that require the associated identity.
```
Comment (Dfinity-Bjoern) [inline docs/modules/ic-identity-guide/pages/auth-how-to.adoc:131]: User number -> ID number.

**Review by Dfinity-Bjoern — COMMENTED** [review]
I think you overwrote all the fixes I made before.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc (hunk 1 above)
Comment (Eve832) [reply docs/modules/ic-identity-guide/pages/auth-how-to.adoc:11]: I know, but until the UI is changed, I have to refer to it. 
And "pseudonym" literally means "fake name" - I can't use it in good conscience.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc (hunk 2 above)
Comment (Eve832) [reply docs/modules/ic-identity-guide/pages/auth-how-to.adoc:13]: So where would the Client Id created for the user live? In an asset canister? Also, I thought that II could work on regular web apps as well as the IC

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc (hunk 3 above)
Comment (Eve832) [reply docs/modules/ic-identity-guide/pages/auth-how-to.adoc:15]: I think Joachim added that. I'll correct it.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc (hunk 5 above)
Comment (Eve832) [reply docs/modules/ic-identity-guide/pages/auth-how-to.adoc:25]: Again, happy to change this when the ICIdentity app incorporates this change.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc (hunk 11 above)
Comment (Eve832) [reply docs/modules/ic-identity-guide/pages/auth-how-to.adoc:103]: ITS LITERALLY IN THE DARN UI!!!

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc (hunk 1 above)
Comment (imageoracle) [reply docs/modules/ic-identity-guide/pages/auth-how-to.adoc:11]: I am just looking at this thread , and see that "user" doesn't sound good, and pseudonym doesn't sound good , what about calling them I iidentities .  Like a double identity.  The people can be ii's, and one person can have another identity.  Plural of I.  
Perhaps people will have iiiiiiiiidenties....

Comment (imageoracle) [reply docs/modules/ic-identity-guide/pages/auth-how-to.adoc:11]: My identity uses an iPhone . The internet computer identity app could be called "ii" ... there's a lot that could be done with the term internet identity and the shortened version of it "ii"

**Review by lsgunnlsgunn — APPROVED** [review]
I agree that using pseudonym doesn't improve clarity and is more likely to lead to confusing misinterpretations.
As for removing "user" from user number, it is still in the UI. I'm actually okay with change it to identification number and treating it as a description what the number is for, in which case having text that does not mirror the UI is probably okay. The rule of thumb though is to use consistent terms in doc and the UI to avoid confusion.

I'd prefer to get this out and iterate on the text over holding this back for this inconsistency.

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc (hunk 6 above)
Comment (lsgunnlsgunn) [reply docs/modules/ic-identity-guide/pages/auth-how-to.adoc:53]: Occurrence references the text displayed in the UI. Has this changed yet?

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc (hunk 7 above)
Comment (lsgunnlsgunn) [reply docs/modules/ic-identity-guide/pages/auth-how-to.adoc:55]: Occurrence references the text displayed in the UI. Has this changed yet?

File: docs/modules/ic-identity-guide/pages/auth-how-to.adoc (hunk 10 above)
Comment (lsgunnlsgunn) [reply docs/modules/ic-identity-guide/pages/auth-how-to.adoc:72]: I'm not sure if this references test in the UI or not.

**Comment by Dfinity-Bjoern** [comment]
I guess we need to pause this until we have some more consistent naming. In the last few days I've received four different requests from Dom (rename it to: ID number, ID anchor, anchor number, identity anchor). I don't know what to do about it. I think we should first fix the naming we want to go with, and then use that.

**Comment by lsgunnlsgunn** [comment]
Can we get a link to the issue or PR where the UI change is tracked so that when the decision has been made, we can return to fix the corresponding doc to match?
(My inclination though would be to use the generic identification number and not invent some new and unfamiliar term for people to have to learn. Simpler is better. But no one listens to me 😉 )

**Comment by imageoracle** [comment]
> I guess we need to pause this until we have some more consistent naming. In the last few days I've received four different requests from Dom (rename it to: ID number, ID anchor, anchor number, identity anchor). I don't know what to do about it. I think we should first fix the naming we want to go with, and then use that.---- identity anchor is perfect because it's opposite ai- artificial intelligence vs identity anchors - real human intelligence .  An anchored identity or identity anchor - those both sound good and safe and solid .

**Comment by kritzcreek** [comment]
We've since reworked much of the terminology, so this PR no longer applies.

**Pull request closed without merging**